lib/host-connection-pool.js

```diff
@@ -169,6 +172,11 @@ HostConnectionPool.prototype._attemptNewConnection = function (callback) {
       c.close();
       return self.emit('open', err);
     }
+    // close the connection if pool was shutdown.
```

Comment:
This seemed to be the most straightforward fix, although there may be a more proactive way to close the connection that is in the process of connecting.